### DISCUSSION TITLE
Hvordan kan vi la folk velge fargetema uavhengig av systeminnstillinger?

### DISCUSSION THREAD

**Discussion by wkillerud** [fremtind/jokul#2806] [Annet]
Jeg så en talk av Rachel Andrews i går på [axe-con med tittelen New CSS with Accessibility in Mind](https://www.deque.com/axe-con/sessions/new-css-with-accessibility-in-mind/). Mest interessant for utviklere, men hun tok opp et punkt jeg vil ta opp litt bredere.

Vi har jo to temaer, lyst og mørkt, som følger systeminnstillingen til brukeren [^1]. Men, det er ikke gitt at akkurat vårt mørke tema er tilgjengelig for alle brukere, selv om de foretrekker mørke temaer. Kanskje er vårt _lyse_ tema bedre for dem, selv om de bruker (og fortsatt vil bruke) mørkt tema på operativsystemet.

I dag har vi ikke noe etablert mønster for å la brukeren selv velge tema, uavhengig av systeminnstilling. Hvordan kan vi gjøre det på en god måte?

[^1]:  For det meste, det kan hardkodes for hele eller deler av sider, men la oss hoppe over det nå.

**Comment by Mikaila94** [comment]
Enig!
Har ikke satt meg selv inn i den tekniske løsningen, men sånt rent designmessig så kan man ta inspirasjon fra andre.
Jeg ser at Google bruker en sånn innstilling-ikon der de får opp en sidebar fra høyre for å endre tema på store skjermer.
På mindre skjermer så ligger den i settings i hamburger-menyen. Kanskje vi kunne hatt en liknende løsning.

**Comment by piofinn** [comment]
Har også tenkt en del på dette. Teknisk bør det være ganske enkelt (sett `data-theme` til brukerens preferanse dersom de har en preferanse, lagre i cookie e.l.), men er nok smart å se litt på best practices for hvordan man viser frem valget på en god måte.

**Edit**: Her er et eksempel fra MDN. Ganske enkelt og oversiktlig, synes jeg!
<img width="175" alt="Skjermbilde 2022-03-21 kl  12 36 32" src="https://user-images.githubusercontent.com/25739615/159253610-cf54af31-03c5-4447-89a1-161b9f857357.png">
